.github/scripts/main/main.sh

```diff
@@ -121,35 +163,88 @@ if [ "$OPAM_TEST" = "1" ]; then
 
   # Compile and run opam-rt
   (set +x ; echo -en "::group::opam-rt\r") 2>/dev/null
-  opamrt_url="https://github.com/ocaml-opam/opam-rt"
-  if [ ! -d $CACHE/opam-rt ]; then
-    git clone $opamrt_url  $CACHE/opam-rt
-  fi
-  cd $CACHE/opam-rt
-  git fetch origin
-  if [ "$GITHUB_EVENT_NAME" = "pull_request" ] && git ls-remote --exit-code origin "$GITHUB_PR_USER/$BRANCH" ; then
-    BRANCH=$GITHUB_PR_USER/$BRANCH
-  fi
-  if git ls-remote --exit-code origin "$BRANCH"; then
-    OPAM_RT_BRANCH=$BRANCH
-  elif [ "$GITHUB_EVENT_NAME" = pull_request ] && git ls-remote --exit-code origin "$GITHUB_BASE_REF"; then
-    OPAM_RT_BRANCH=$GITHUB_BASE_REF
-  else
-    OPAM_RT_BRANCH=master
-  fi
-  if git branch | grep -q "$OPAM_RT_BRANCH"; then
-    git checkout "$OPAM_RT_BRANCH"
-    git reset --hard "origin/$OPAM_RT_BRANCH"
-  else
-    git checkout -b "$OPAM_RT_BRANCH" "origin/$OPAM_RT_BRANCH"
-  fi
+  prepare_project "https://github.com/ocaml-opam/opam-rt" "opam-rt"
 
-  test -d _opam || opam switch create . --no-install --formula '"ocaml-system"'
-  eval $(opam env)
-  opam pin $GITHUB_WORKSPACE -yn --with-version to-test
   # opam lib pins defined in opam-rt are ignored as there is a local pin
   opam pin . -yn --ignore-pin-depends
-  opam install opam-rt --deps-only opam-devel.to-test
-  make || { opam reinstall opam-client -y; make; }
+  opam install opam-rt --deps-only opam-devel
+  opam exec -- make || { opam reinstall opam-client -y; opam exec -- make; }
   (set +x ; echo -en "::endgroup::opam-rt\r") 2>/dev/null
 fi
+
+test_project () {
+  url=$1
+  project=$2
+
+  (set +x; echo -en "::group::depends-$project\r") 2>/dev/null
+  set +e
```

Comment:
why is this added? If it's just to get the return code of a failing command this can be done easily with `cmd || ret=$?` instead of the current `cmd ; ret=$?`

Case-in-point: the 5.3.0 job currently doesn't do anything and does not report the failure in the script:
```
Error:  Package opam-state has no version 2.4.1.
Error:  Package opam-solver has no version 2.4.1.
Error:  Package opam-repository has no version 2.4.1.
Error:  Package opam-installer has no version 2.4.1.
Error:  Package opam-format has no version 2.4.1.
Error:  Package opam-devel has no version 2.4.1.
Error:  Package opam-core has no version 2.4.1.
Error:  Package opam-client has no version 2.4.1.
```